packages/jest-core/src/collectHandles.ts

```diff
@@ -62,12 +62,16 @@ export default function collectHandles(): HandleCollectionResult {
       triggerAsyncId,
       resource: {} | NodeJS.Timeout,
     ) {
+      // Skip resources that should not generally prevent the process from
+      // exiting, not last a meaningfully long time, or otherwise shouldn't be
+      // tracked.
       if (
         type === 'PROMISE' ||
         type === 'TIMERWRAP' ||
         type === 'ELDHISTOGRAM' ||
         type === 'PerformanceObserver' ||
-        type === 'RANDOMBYTESREQUEST'
+        type === 'RANDOMBYTESREQUEST' ||
+        type === 'DNSCHANNEL'
```

Comment:
It occurred to me while adding this that the way we bail out early here can break the chain of indirect async resource tracking: https://github.com/facebook/jest/blob/ae3fc943d48a8b690e7cd5872f652c6aaef91ddb/packages/jest-core/src/collectHandles.ts#L80-L83

I don’t think it makes sense to change in this PR (it could be pretty high impact), but I wonder if we should not exit early here, but instead track these with `isActive` set to always return `false`.

Reply:
I'm open to a PR, but I don't really understand how it might break something? I'm sure I'll be convinced, tho! 😀

Reply:
I might just be being paranoid! My worry is that this will act too aggressively and cause us to warn on Jest internals or other things that should be ok. Will give this a try and see what happens.